tests/rptest/tests/cluster_self_config_test.py

```diff
@@ -0,0 +1,100 @@
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=1)
+    @matrix(cloud_storage_enable_remote_read=[True, False],
+            cloud_storage_enable_remote_write=[True, False])
+    def test_s3_self_config(self, cloud_storage_enable_remote_read,
+                            cloud_storage_enable_remote_write):
+        """
+        Verify that cloud_storage_url_style self configuration occurs for the s3_client
+        when it is not specified. There aren't any endpoints for testing this, so
+        it will be manually checked for from the logs.
+        """
+
+        config = self.admin.get_cluster_config()
+
+        # Even after self-configuring, the cloud_storage_url_style setting will
+        # still be left unset at the cluster config level.
+        assert config['cloud_storage_url_style'] == None
+
+        def self_config_start_in_logs(node):
+            client_self_configuration_start_string = 'Client requires self configuration step'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_start_string):
+                m = re.search(client_self_configuration_start_string,
+                              log.strip())
+                if m:
+                    return True
+            return False
+
+        def self_config_result_from_logs(node):
+            client_self_configuration_complete_string = 'Client self configuration completed with result'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_complete_string):
+                m = re.search(
+                    client_self_configuration_complete_string + r' (\{.*\})',
+                    log.strip())
+                if m:
+                    return (True, m.group(1))
+            return (False, None)
+
+        def self_config_default_in_logs(node):
+            client_self_configuration_default_string = r'Can\'t self-configure S3 Client'
+            for log in self.log_searcher._capture_log(
+                    node, client_self_configuration_default_string):
+                m = re.search(client_self_configuration_default_string,
+                              log.strip())
+                if m:
+                    return True
+            return False
+
+        for node in self.redpanda.nodes:
+            #Assert that self configuration started.
+            assert self_config_start_in_logs(node)
+
+            #If neither remote_read or remote_write are enabled, check for the "defaulting" output
+            if not cloud_storage_enable_remote_read and not cloud_storage_enable_remote_write:
+                assert self_config_default_in_logs(node)
+
+            #Assert that self configuration returned a result.
+            self_config_result = self_config_result_from_logs(node)
+            assert self_config_result[0]
+
+            #Currently, virtual_host will succeed in all cases with MinIO.
```

Comment:
Is the same true when running in CDT? I didn't follow the prior work in client configuration, but I'm wondering if the other cases are verifiable

Reply:
I think `virtual_host` will always succeed since it is the first tested- but it would probably be better here to have 

```python
self_config_expected_results = ['{s3_self_configuration_result: {s3_url_style: virtual_host}}',
                                                         '{s3_self_configuration_result: {s3_url_style: path}}'
assert self_config_result[1] in self_config_expected_results
```

for coverage's sake.

I wish there were an easy way to "flip" MinIO to allow/disallow one style or the other for complete coverage, but I don't think we can do so from the ducktape layer.